### DISCUSSION TITLE
failed to publish artifacts

### DISCUSSION THREAD

**Discussion by andig** [goreleaser/goreleaser#1421] [Q&A] [answered]
Running locally, I'm unable to push to github but cannot find any reason:

````
   • publishing       
      • blobs            
         • pipe skipped              error=Blob section is not configured
      • http upload      
         • pipe skipped              error=uploads section is not configured
      • artifactory      
         • pipe skipped              error=artifactory section is not configured
      • docker images    
      • snapcraft packages
      • github/gitlab/gitea releases
         • token type                type=github
         • creating or updating release repo=andig/evcc tag=0.2.1
         • release updated           url=
   ⨯ release failed after 8.96s error=github/gitlab/gitea releases: failed to publish artifacts: POST https://api.github.com/repos/andig/evcc/releases: 404 Not Found []
````

[goreleaser.txt](https://github.com/goreleaser/goreleaser/files/4427285/goreleaser.txt)

Most certainly my fault but I have no idea where to look?

**Comment by caarlos0** [comment]
the 404 is very weird... I would say maybe something on github side... can you try again?

**Reply by andig** [reply]
Did- same thing, no further info with `--debug`. I can `GET` it and a simple `POST` without body gives:

```
{
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3/repos/releases/#create-a-release"
}
```

**Reply by caarlos0** [reply]
seems like something on github side, maybe open a support case with them?

For example, on a repo of mine, If I try an empty post I get I totally different message:

```
λ curl -X POST -H "Authorization: token $GITHUB_TOKEN" https://api.github.com/repos/caarlos0/test/releases -d'{}'
{
  "message": "Invalid request.\n\n\"tag_name\" wasn't supplied.",
  "documentation_url": "https://developer.github.com/v3/repos/releases/#create-a-release"
}
```

**Reply by andig** [reply]
mhhm. I'll retry with new token:

```
❯ curl -X POST -H "Authorization: token $GITHUB_TOKEN" https://api.github.com/repos/andig/evcc/releases -d'{}'

{
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3/repos/releases/#create-a-release"
}
```

**Reply by andig** [reply]
Ahh, that's tricky:

<img width="756" alt="Screenshot 2020-04-03 at 15 37 16" src="https://user-images.githubusercontent.com/184815/78366627-0a4a4980-75c1-11ea-816a-47f10e863841.png">

The UI seems to indicate that this is *only* relevant for private repos. It seems though that I need to check the repo section even for a public repo. Doing so I now get the same result as you.

Are you able to verify that?

**Reply by andig** [reply]
More precisely, the `public_repo` permission is required.

**Reply by caarlos0** [reply]
oh yes, you need the public repo permission... but AFAIK when you didn't had it on public repos the error was different...

**Reply by caarlos0** [reply]
anyway, glad you figure it out :D

**Comment by caarlos0** [comment, marked as answer]
need to have `public_repo` permission on the github token being used.

check https://github.com/goreleaser/goreleaser/discussions/1421#discussioncomment-3118

**Reply by andig** [reply]
Much appreciated, thank you!
